### DISCUSSION TITLE
Broken query and sorting not working when using custom column type

### DISCUSSION THREAD

**Discussion by DamianGosciewski** [rappasoft/laravel-livewire-tables#1229] [Q&A] [answered]
Hi, just found this project recently, fantastic job. To the point:

What I wanted to achieve is **display name in table column with the link pointing to the details page**.

My first thought was to use `Rappasoft\LaravelLivewireTables\Views\Columns\LinkColumn` like this:
```PHP
public function columns(): array
{
    return [
        LinkColumn::make('Name', 'name')
            ->title(fn($row) => $row->name)
            ->location(fn($row) => route('some.route.edit', [
                'id' => $row->id
            ]))
            ->sortable(),
    ];
}
```

Problem is that constructor uses `label()` method (which skips query select and breaks default sorting):
```PHP
public function __construct(string $title, string $from = null)
{
    parent::__construct($title, $from);

    $this->label(fn () => null);
}
``` 
What this method does is resetting `$field`:

```PHP
public function label(callable $callback): self
{
    $this->from = null;
    $this->field = null;
    $this->labelCallback = $callback;

    return $this;
}
```

I guess this is intended, I've done some searching ([#946](https://github.com/rappasoft/laravel-livewire-tables/issues/946#issuecomment-1278902224), [#961](https://github.com/rappasoft/laravel-livewire-tables/issues/961#issuecomment-1340869410)) and found solutions like using `format()` on `Column`, but it feels like custom columns concept is misleading.

There is also other problem - query not having field (reset by `label()`) and `builder()` method has to be implemented like:

```PHP
public function builder(): Builder
{
    return SomeClass::query()->select();
}
```
Otherwise `$row` won't have required fields and title method will throw error when trying to get row data like so:
```PHP
LinkColumn::make('Name', 'name')
            ->title(fn($row) => $row->name)
````

Is it a bug or maybe I do something wrong?

**Comment by lrljoe** [comment, marked as answer]
You can use setAdditionalSelects() in your configure() method to add extra fields to the query.

Sortable labels fix is in an upcoming (likely next) release.
